sql/server.go

```diff
@@ -155,13 +158,29 @@ func (p parameters) Arg(i int) (parser.Datum, bool) {
 // exec executes the request. Any error encountered is returned; it is
 // the caller's responsibility to update the response.
 func (s *Server) exec(req driver.Request) (resp driver.Response, err error) {
-
 	// Pick up current session state.
 	// The request user is validated in ServeHTTP. Even in insecure mode,
 	// it is guaranteed not to be empty.
 	planMaker := planner{user: req.GetUser()}
 	defer func() {
 		// Update session state even if an error occurs.
+		if err != nil && planMaker.txn != nil {
+			// If we hit an error and there is a pending transaction, rollback
+			// the transaction before returning. The client does not have to
+			// deal with cleaning up transaction state.
+			if rollbackErr := planMaker.txn.Rollback(); rollbackErr != nil {
```

Comment:
We do not differentiate between abort and rollback. I looked briefly around, can you point out the article that talks about aborted transactions hanging around.

I suppose postgres allows you to ROLLBACK to a savepoint which is the main difference between ABORT and ROLLBACK.

Should I bundle the two errors into one?